src/App.tsx

```diff
@@ -1,26 +1,182 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  ChangeEvent,
+  KeyboardEvent,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { FilterStatusType, Todo } from './types/Todo';
+import * as tadoService from './api/todos';
+import { Footer } from './component/Footer';
+import { Error } from './component/Error';
+import { ListComponent } from './component/ListComponent';
+import { ErrorMessages, errorMessages } from './types/err';
 
-const USER_ID = 0;
+function filterTodos(todos: Todo[], filter: FilterStatusType) {
+  switch (filter) {
+    case FilterStatusType.All:
+      return todos;
+    case FilterStatusType.Active:
+      return todos.filter(todo => !todo.completed);
+    case FilterStatusType.Completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterStatusType>(
+    FilterStatusType.All,
+  );
+  const [query, setQuery] = useState('');
+  const [errorMessage, setErrorMessage] = useState<ErrorMessages | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isLoadingTodos, setIsLoadingTodo] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const handleError = (message: ErrorMessages) => {
+    setErrorMessage(message);
+
+    setTimeout(() => {
+      setErrorMessage(null);
+    }, 3000);
+  };
+
+  useEffect(() => {
+    tadoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        handleError(errorMessages.load);
+      });
+  }, []);
+
+  const handleDelete = (todoId: number) => {
+    setIsLoadingTodo(prev => [...prev, todoId]);
+
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+
+    tadoService
+      .deleteTodos(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        handleError(errorMessages.delete);
+      })
+      .finally(() => {
+        setIsLoadingTodo(prev => prev.filter(id => id !== todoId));
+      });
+  };
+
+  const handleQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const handleAddTodo = (event: KeyboardEvent<HTMLInputElement>): void => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+
+      if (query.trim() === '') {
+        handleError('Title should not be empty');
```

Comment:
let's create enum with error messages and use its values instead of magic string